File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int n = nums.size();
        vector<int> LIS(n, 1);
        int res = 1;
        for(int i = n - 1; i > -1; --i) {
            int maxLISi = 1;
            for(int j = i + 1; j < n; ++j) {
                // cout << nums[j] << " " << nums[i] << endl;
                if(nums[j] > nums[i]) {
                    maxLISi = max(maxLISi, LIS[j] + 1);
                }
            }
            LIS[i] = maxLISi;
            res = max(res, LIS[i]);
        }
        return res;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numset = set(nums)
        res = 0
        for num in nums:
            if(num + 1 not in nums):
                seq = 0
                i = 0
                while(num - i in nums):
                    seq += 1
                    i += 1
                res = max(res, seq)
        
        return res